zcash_primitives/src/transaction/mod.rs

```diff
@@ -748,6 +779,44 @@ impl Transaction {
             sprout_bundle: None,
             sapling_bundle,
             orchard_bundle,
+            #[cfg(all(zcash_unstable = "nu7", feature = "zip-233"))]
+            zip233_amount,
+            #[cfg(zcash_unstable = "zfuture")]
+            tze_bundle,
+        };
+
+        Ok(Self::from_data_v5(data))
+    }
+
+    #[cfg(zcash_unstable = "nu7")]
+    fn read_v6<R: Read>(mut reader: R, version: TxVersion) -> io::Result<Self> {
+        let (consensus_branch_id, lock_time, expiry_height) =
+            Self::read_v5_header_fragment(&mut reader)?;
+        let transparent_bundle = Self::read_transparent(&mut reader)?;
+        let sapling_bundle = sapling_serialization::read_v5_bundle(&mut reader)?;
+        let orchard_bundle = orchard_serialization::read_v5_bundle(&mut reader)?;
+
+        #[cfg(feature = "zip-233")]
+        let zip233_amount = if version == TxVersion::V6 {
```

Comment:
This should not be conditional on `version == TxVersion::V6`, because this method is by definition for the v6 format. (Also if it were reused for v7, say, this would be wrong.)